turnwithus/lytics.com
# LYT-62: add search functionality to integration list

* adds script to filter integration partner items
* updates markup in both `.hbs` and `.twig` files to make that script easier
* adds styles for showing/hiding filtered items
* adds styles to toggle search icon with X when searching
* adds a polyfill for `Element.closest` api (needed for all versions of IE)

**To test:**
* navigate to the integrations landing page
* start searching for partners
* items should filter as you type
* magnifying glass should change to X
* clicking X should clear input, return all items to screen, return magnifying glass to input
* deleting search term/manually clearing input should return all items to screen & return magnifying glass to input
* search term should match any string contained within partner titles (Urban Airship should be a match to `urban`, `airship`, `ur`, `ban`, `sh`, `p`, etc)

**TODO:**
* awaiting response on how to handle no results (current site returns a form); functionality implemented here only adds a `filtered--no-results` class to the parent `.IntegrationList` and nothing else. Assumed we could use that to toggle the form if it's needed. If not, we can build and display a message.
